lib/nokogiri/xml/node_set.rb

```diff
@@ -293,7 +293,11 @@ def == other
       # Returns a new NodeSet containing all the children of all the nodes in
       # the NodeSet
       def children
-        inject(NodeSet.new(document)) { |set, node| set += node.children }
```

Comment:
I'm not sure I understand why this change is being made. Is it for performance? If so, can you share your benchmarks?

I'm asking only because it seems out of place in this patchset.